prac_05/electricity_bill_estimator_v2.py

```diff
@@ -10,18 +10,29 @@
 
 # Version II
 
-TARIFF_11 = 0.244618
-TARIFF_31 = 0.136928
+TARIFFS = {11: 0.244618, 31: 0.136928}
+
 print('Welcome to the Electricity bill estimator 2.0')
+for key in TARIFFS:
+    print("Tarriff {} is {} cents per hour".format(key, TARIFFS[key]))
+
+correct_input = False
+
 
-tariff = int(input("Which tariff? 11 or 31: "))
-if tariff == 11:
-    tariff = TARIFF_11
-else:
-    tariff = TARIFF_31
+while not correct_input:
+    tariff = -1
+    try:
+        tariff = int(input("Which tariff: "))
+    except ValueError:
+        print("Please enter an integer")
+        continue
+    if tariff not in TARIFFS:
+        print("Tariff not found, try again")
+    else:
+        correct_input = True
 
 daily_use = float(input("Enter daily use in kWh: "))
 billing_days = int(input("Enter number of billing days: "))
 total_bill = (tariff * daily_use) * billing_days
```

Comment:
Currently total_bill is using the user input 'tariff' to multiply, not pulling the tariff from the dictionary.
try replacing tariff with: TARIFFS[tariff] or TARIFFS.get(tariff)

Reply:
Thanks Steve! 
Yep I caught the tarriff error over the weekend, when my bill was massive; interesting example of a logic error that won't necessarily get picked up without testing I guess. As for the main() refactor I'm never sure what the autograder wants so I've been trying to just leave my solutions as close to the prac files as possible. Cheers mate, happy coding.